File: sorting/merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        vector<int> vec;  // Temporary vector

        // Copy first m elements from nums1
        for (int i = 0; i < m; i++) {
            vec.push_back(nums1[i]);
        }

        // Copy first n elements from nums2 (Fixed the loop condition)
        for (int i = 0; i < n; i++) {
            vec.push_back(nums2[i]);
        }

        // Sort the merged vector
        sort(vec.begin(), vec.end());

        // Copy sorted elements back to nums1
        for (int i = 0; i < m + n; i++) {
            nums1[i] = vec[i];
        }
    }
};

File: 88-merge-sorted-array/merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int i=m-1;
        int j=n-1;
        int fin=nums1.length-1;
        while(j>=0){
            if(i>=0&&nums1[i]>=nums2[j]){
                nums1[fin]=nums1[i];
                System.out.println(nums1[fin]);
                i--;
            }else{
                nums1[fin]=nums2[j];
                System.out.println(nums2[j]);
                j--;
            }
            fin--;
        }
        
    }
}